### ISSUE TITLE
Wiring information doesn't work with V3 of the Wemos D1

### ISSUE THREAD

**Issue by spidurcam** [unixko/MitsuCon#15]
Great work on this stuff, really appreciate the work it is awesome.

In the latest version of the Wemos D1 mini they have removed the series resistor from the transmit line on the serial to USB IC. This means wiring it up as the readme shows doesn't work and you get no data from the heatpump. The work around I have done currently is to lift the TX and RX pins on the UART to USB IC, however it means I can no longer program the device.

Can we add support for using different pins for UART (maybe software UART required?). And maybe add a caution to the readme about this latest verison?

![wemos comparison](https://user-images.githubusercontent.com/61809161/75933956-5aa18f00-5ecf-11ea-8fe4-c5f0d6df3ae6.png)

**Comment by unixko** [comment]
Thanks for information. I noticed this too but I can't figure out what happen so I just stick with V2 board. About softserial it need more work from upstream library https://github.com/SwiCago/HeatPump/issues/136 .

**Comment by danolson1** [comment]
I had the same problem when using a Wemos D1 mini clone from Amazon (B081PX9YFV). I took a look at the serial signal and noticed that the negative signal on the Rx pin was only getting down to 1.8V. I added a 1k ohm resistor between the Rx and ground pins on my level converter, and that solved the problem. I tested a few different pull-down resistor values, and anything from about 100 ohms - 1.5k ohms seems to work.

One benefit of this approach is that you can still program the device from the USB port if needed.

![Artboard 1](https://github.com/unixko/MitsuCon/assets/5619720/a23c82b5-41cf-46a7-8ac0-cb0bb916c33d)

**Comment by smuskiew** [comment]
@danolson1 Thanks for this - I was having the same problem with a similar ESP board.

Can you clarify the pinouts?  I'm looking at your pic and it isn't entirely clear but for what you're calling the "Rx" pin where the resistor goes did you mean it is for the Rx signal line on the ESP board side (which would actually be the Tx pin on the Heat pump CN105 connector side).  That's what it looks like from the picture, unless I'm just confusing what I see.

So the pinout looks to be:
CN 105 Tx (Black wire) -> HV2  <> LV2 -> ESP Rx (aqua wire)
CN 105 Rx (red wire) -> HV1 <> LV1 -> ESP Tx (blue wire)
CN 105 5V (yellow wire) -> HV <> LV  -> ESP 3.3v (white wire)
CN 105 GND (green wire) -> GND <> GND -> ESP GND (black wire)
1k resistor between HV2 (CN105 Tx) and GND on HV side of level converter
plus the CN105 5V connects to the ESP 5V

Is that correct or am I mixing up the Rx and Tx lines?

**Comment by danolson1** [comment]
Yes. That's correct. The scope readings were taken on the CN105 side.

So the "Tx line" scope readings were taken from the red wire (HV1) going to the CN105 connector, and the "Rx line" scope readings were taken from the black wire (HV2) going to the CN105 connector

**Comment by matthewbarr** [comment]
Is there anyway to test this quickly w/o a scope?  Supposedly - the board diagram *claims* to have the resistor needed, but the actual picture of what was sent was different.  I want to make sure I'm going to be OK before I bother mounting into a HVAC unit.

I may have already missed the return window, but.. worth finding out before hand.

**Comment by danolson1** [comment]
A scope is pretty useful for this sort of thing.

You could build your circuit on a breadboard at first. Then it's easy to
test different pull down resistors.

-Dan

**Comment by matthewbarr** [comment]
> A scope is pretty useful for this sort of thing. You could build your circuit on a breadboard at first. Then it's easy to test different pull down resistors.

I've tested w/ a multi meter.  

By setting the pin 1 & 3 to pinmode function 3 in Arduino IDE, and then setting it as an output, I get 3.3v when high and 0 when low.  (basically, the opposite of the LED)

No resistors, no extra board.

Will the D1 work correctly w/ the Split.. is the key question.  I suspect the answer is yes, since it seems like it has the same behavior between both TX & RX, which would align better w/ the board V2 (resistor on both TX & RX)  vs v3(which has only one resistor).

**Comment by jojocat** [comment]
> I had the same problem when using a Wemos D1 mini clone from Amazon (B081PX9YFV). I took a look at the serial signal and noticed that the negative signal on the Rx pin was only getting down to 1.8V. I added a 1k ohm resistor between the Rx and ground pins on my level converter, and that solved the problem. I tested a few different pull-down resistor values, and anything from about 100 ohms - 1.5k ohms seems to work.

Thanks so much for this - the 1k ohm resistor worked great for me with this same clone board.

**Comment by superbitcoin77** [comment]
> > I had the same problem when using a Wemos D1 mini clone from Amazon (B081PX9YFV). I took a look at the serial signal and noticed that the negative signal on the Rx pin was only getting down to 1.8V. I added a 1k ohm resistor between the Rx and ground pins on my level converter, and that solved the problem. I tested a few different pull-down resistor values, and anything from about 100 ohms - 1.5k ohms seems to work.
> 
> Thanks so much for this - the 1k ohm resistor worked great for me with this same clone board.

I have the same problem, I can control the heatpump but I don't receive anything, D1mini purchased on Amazon. 
Are you telling me that you put the resistor without a converter?

**Comment by jojocat** [comment]
> > > I had the same problem when using a Wemos D1 mini clone from Amazon (B081PX9YFV). I took a look at the serial signal and noticed that the negative signal on the Rx pin was only getting down to 1.8V. I added a 1k ohm resistor between the Rx and ground pins on my level converter, and that solved the problem. I tested a few different pull-down resistor values, and anything from about 100 ohms - 1.5k ohms seems to work.
> > 
> > 
> > Thanks so much for this - the 1k ohm resistor worked great for me with this same clone board.
> 
> I have the same problem, I can control the heatpump but I don't receive anything, D1mini purchased on Amazon. Are you telling me that you put the resistor without a converter?

No I was using the converter initially but had the issue and resolved it by also adding the resistor per the photos by danolson1.
